src/dotnet/APIView/APIViewWeb/Repositories/PullRequestManager.cs

```diff
@@ -0,0 +1,182 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text.Json;
+using System.Threading.Tasks;
+using ApiView;
+using APIViewWeb.Models;
+using APIViewWeb.Repositories;
+using APIViewWeb.Respositories;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Octokit;
+
+namespace APIViewWeb.Repositories
+{
+    public class PullRequestManager
+    {
+        static readonly HttpClient devopsClient = new();
+        static readonly GitHubClient githubClient = new(new Octokit.ProductHeaderValue("apiview"));
+        static readonly string ARTIFACT_GET_URL = "https://dev.azure.com/azure-sdk/public/_apis/build/builds/{buildId}/artifacts?artifactName={artifactName}&api-version=4.1";
+
+        private readonly ReviewManager _reviewManager;
+        private readonly CosmosPullRequestsRepository _pullRequestsRepository;
+
+        public PullRequestManager(
+            ReviewManager reviewManager,
+            CosmosReviewRepository reviewsRepository,
+            CosmosPullRequestsRepository pullRequestsRepository,
+            IConfiguration configuration
+            )
+        {
+            _reviewManager = reviewManager;
+            _pullRequestsRepository = pullRequestsRepository;
+            githubClient.Credentials = new Credentials(configuration["github-access-token"]);
+        }
+
+        // API change detection for PR will pull artifact from devops artifact
+        // This will be changed to a different component in the future based on the implementation for SDK automation
+        public async Task DetectApiChanges(string buildId, string artifactName, string filePath, int prNumber, string commitSha, string language)
+        {
+            TelemetryClient telemetryClient = new(TelemetryConfiguration.CreateDefault());
+            var requestTelemetry = new RequestTelemetry { Name = "Detecting API changes for PR: " + prNumber };
+            var operation = telemetryClient.StartOperation(requestTelemetry);
+            try
+            {
+                if (await IsShaAlreadyProcessed(language, prNumber, commitSha))
+                {
+                    return;
+                }
+
+                var pullRequestModel = await _pullRequestsRepository.GetPullRequestAsync(prNumber, language);
+                if (pullRequestModel == null)
+                {
+                    pullRequestModel = new PullRequestModel()
+                    {
+                        Language = language,
+                        CommitSha = commitSha,
+                        PullRequestNumber = prNumber
+                    };
+                }
+                else
+                {
+                    pullRequestModel.CommitSha = commitSha;
```

Comment:
Do we need to keep a list of commitSha's or just the latest?

Reply:
I understand this can still lead to an issue if pipeline is run with old commit ID many times. But adding all commit ID will increase cosmos usage drastically.

Reply:
I have changed to take list of Commit Sha since this can lead to duplicate run with same commit ID .